code/game/atoms_movable.dm

```diff
@@ -158,7 +158,8 @@
 	//use a modified version of Bresenham's algorithm to get from the atom's current position to that of the target
 
 	src.throwing = 1
-	SpinAnimation(5, 1)
+	if(!istype(target, /obj/singularity)) // turns out 1000+ spinning objects being thrown at the singularity creates lag - Iamgoofball

```

Comment:
Yeah this is pretty awful, anyone have a better solution? A second throw proc without spin perhaps?